# HG changeset patch
# User Torsten Lilge <ttl-octave@mailbox.org>
# Date 1552454653 -3600
#      Wed Mar 13 06:24:13 2019 +0100
# Node ID 9fa577cf9f4e8460aad6c685ed52fbaa9697b1a6
# Parent  a1acd11bad1913dea306f471ce3b6c1597b2222c
prevent unnamed editor tab from being closed by rmdir (bug #55888)

* file-editor.cc (handle_dir_remove): do not check an editor tab with
  empty, i.e. invalid, filename

diff --git a/libgui/src/m-editor/file-editor.cc b/libgui/src/m-editor/file-editor.cc
--- a/libgui/src/m-editor/file-editor.cc
+++ b/libgui/src/m-editor/file-editor.cc
@@ -2424,16 +2424,19 @@ namespace octave
     // Have all file editor tabs signal what their filenames are.
     m_editor_tab_map.clear ();
     emit fetab_file_name_query (nullptr);
 
     // Loop over all open files and pick those within old_dir
     for (auto p = m_editor_tab_map.cbegin ();
          p != m_editor_tab_map.cend (); p++)
       {
+        if (p->first.isEmpty ())
+          continue;   // Nothing to do, no valid file name
+
         QString rel_path_to_file = old_dir.relativeFilePath (p->first);
         if (rel_path_to_file.left (3) != QString ("../"))
           {
             // We directly go down from old_dir to reach our file: Our
             // file is included in the removed/renamed diectory.
             // Thus delete it.
             m_no_focus = true;  // Remember for not focussing editor
             file_editor_tab *editor_tab
